src/app/screens/signatureRequest/index.tsx

```diff
@@ -289,9 +289,19 @@ function SignatureRequest(): JSX.Element {
           finalizeMessageSignature({ requestPayload: request, tabId: +tabId, data });
         }
       }
-    } catch (e) {
+    } catch (e: any) {
       console.error(e);
-      setIsTxRejected(true);
+
+      if (e.name === 'LockedDeviceError') {
```

Comment:
we should probably create an enum with possible errors and error codes if we don't have one yet

Reply:
Yeah agree, I added that as a point for the refactoring task https://linear.app/xverseapp/issue/ENG-2537/refactor-ledger-related-logic, I'm planning to revamp the ledger errors handling in scope of it